core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -152,43 +140,38 @@ class LoadBalancerService(
      * Creates an activation entry and insert into various maps.
      */
     private def setupActivation(action: ExecutableWhiskAction, activationId: ActivationId, namespaceId: UUID, invokerName: String, transid: TransactionId): ActivationEntry = {
-        // either create a new promise or reuse a previous one for this activation if it exists
+
         val timeout = action.limits.timeout.duration + activeAckTimeoutGrace
-        val entry = activationById.getOrElseUpdate(activationId, {
-            // Install a timeout handler for the catastrophic case where an active ack is not received at all
-            // (because say an invoker is down completely, or the connection to the message bus is disrupted) or when
-            // the active ack is significantly delayed (possibly dues to long queues but the subject should not be penalized);
-            // in this case, if the activation handler is still registered, remove it and update the books.
-            actorSystem.scheduler.scheduleOnce(timeout) {
-                processCompletion(Left(activationId), transid, forced = true)
-            }
+        // Install a timeout handler for the catastrophic case where an active ack is not received at all
+        // (because say an invoker is down completely, or the connection to the message bus is disrupted) or when
+        // the active ack is significantly delayed (possibly dues to long queues but the subject should not be penalized);
+        // in this case, if the activation handler is still registered, remove it and update the books.
+
+        actorSystem.scheduler.scheduleOnce(timeout) {
+            processCompletion(Left(activationId), transid, forced = true)
+        }
 
-            ActivationEntry(activationId, namespaceId, invokerName, Instant.now(Clock.systemUTC()), Promise[Either[ActivationId, WhiskActivation]]())
-        })
+        val entry = ActivationEntry(activationId, namespaceId, invokerName, Instant.now(Clock.systemUTC()),
+            Promise[Either[ActivationId, WhiskActivation]]())
 
         // add the entry to our maps, for bookkeeping
-        activationByInvoker.getOrElseUpdate(invokerName, new TrieSet[ActivationEntry]).put(entry, {})
-        activationByNamespaceId.getOrElseUpdate(namespaceId, new TrieSet[ActivationEntry]).put(entry, {})
+        loadBalancerData.putActivation(entry)
```

Comment:
It was not that trivial to map on the None value that is returned by `activationsById.put(entry.id, entry)` if the entry did not exist, thus we have decided to leave it as it is. The idea was to have a similar implementation as for the  `removeActivation`. e.g. it would have looked somehow like this:
```
        activationsById.put(entry.id, entry).orElse {
            activationByNamespaceId.getOrElseUpdate(entry.namespaceId, new TrieSet[ActivationEntry]).put(entry, {})
            activationByInvoker.getOrElseUpdate(entry.invokerName, new TrieSet[ActivationEntry]).put(entry, {})
            Option[ActivationEntry](entry)
        }
```